compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -530,6 +530,11 @@ class SpaceEngine(using Context) extends SpaceLogic {
     else convertConstantType(tp1, tp2)
   }
 
+  private val isSubspaceCache = mutable.HashMap.empty[(Space, Space, Context), Boolean]
```

Comment:
The cache should not outlive a compilation run otherwise it will leak memory.

I believe `Context` has a mechanism to store such attributes. E.g. https://github.com/lampepfl/dotty/blob/4437469922270098ca980808ad2c6eee92ab445b/compiler/src/dotty/tools/dotc/core/Contexts.scala#L197-L204

Reply:
But this looks like a field in `SpaceEngine` and a new one is instantiated for every `match` expression. Or am I missing something here?

Reply:
I missed that. I thought it was in the object `SpaceEngine`. Then ignore my comment